lib/apipie_bindings/action.rb

```diff
@@ -57,35 +55,40 @@ def find_route(params={})
       return suitable_route
     end
 
-    def validate!(params)
-      missing_params = missing_params(api_params_tree { |par| par.required? }, params_tree(params))
-      raise ApipieBindings::MissingArgumentsError.new(missing_params) unless missing_params.empty?
-    end
-
-    def api_params_tree(&block)
-      ApipieBindings::Utilities.params_hash_tree(self.params, &block)
-    end
-
-    def params_tree(params)
-      params.inject([]) do |tree, (key, val)|
-        subtree = val.is_a?(Hash) ? { key.to_s => params_tree(val) } : key.to_s
-        tree << subtree
-        tree
-      end
+    def validate!(parameters)
+      errors = validate(params, parameters)
```

Comment:
what about using `values` instead of `parameters`. the `params` vs `parameters` might be a bit misleading